### PR TITLE
bpo-38471: Fix _ProactorDatagramTransport close() behaviour

### PR DESCRIPTION
**Pull request by primal100**
Fixes an issue where if the _ProactorDatagramTransport close method is called while there is data in the write buffer, the data is not sent and connection_lost is not called. Behaviour now matches _SelectorDatagramTransport.

https://bugs.python.org/issue38471

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@primal100

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by asvetlov** [comment]
NEWs entry is required: https://devguide.python.org/committing/#what-s-new-and-news-entries

**Comment by primal100** [comment]
The news entry was added

File: Doc/whatsnew/3.8.rst [hunk 1]
```diff
@@ -47,14 +47,6 @@
 This article explains the new features in Python 3.8, compared to 3.7.
 For full details, see the :ref:`changelog <changelog>`.
 
-Prerelease users should be aware that this document is currently in
```
Comment (asvetlov) [inline Doc/whatsnew/3.8.rst:50]: This change is not related to the issue

File: Doc/whatsnew/3.8.rst [hunk 2]
```diff
@@ -584,6 +588,23 @@ The :func:`ast.parse` function has some new flags:
 asyncio
 -------
 
+Running ``python -m asyncio`` launches a natively async REPL.  This allows rapid
```
Comment (asvetlov) [inline Doc/whatsnew/3.8.rst:591]: The same

File: Lib/test/support/__init__.py [hunk 3]
```diff
@@ -1068,7 +1068,7 @@ def change_cwd(path, quiet=False):
     """
     saved_dir = os.getcwd()
     try:
-        os.chdir(path)
+        os.chdir(os.path.realpath(path))
```
Comment (asvetlov) [inline Lib/test/support/__init__.py:1071]: Why is this update needed?

**Review by asvetlov — CHANGES_REQUESTED** [review]

**Comment by primal100** [comment]
Github is showing three commits that aren't mine this pull request. I have no idea why or how to fix it, so will just start over with a new fork and pull request.

**Comment by asvetlov** [comment]
You can force-push (`git push -f`) here without recreating the PR

**Pull request closed without merging**